wgpu-hal/src/gles/adapter.rs

```diff
@@ -365,8 +432,19 @@ impl crate::Adapter<super::Api> for super::Adapter {
                 copy_fbo: gl
                     .create_framebuffer()
                     .map_err(|_| crate::DeviceError::OutOfMemory)?,
+                shader_clear_program,
+                shader_clear_program_color_uniform_location,
                 zero_buffer,
                 temp_query_results: Vec::new(),
+                state: Mutex::new(super::QueueState {
```

Comment:
is this necessary to be a part of the fastclear workaround? I'd like to discuss this separately

Reply:
The mutex made it possible to modify the state ( mutable access ) while also having a reference to `self.shared.context` ( immutable access ), which rust didn't like without the mutex. There might be another way to get around that. I'll see if there's an alternative I could get working. We might just need to not do `let gl = &self.shared.context` at the top of everything so that we can still mutably access our internal state.

Reply:
Why does the queue state need to be shared with anything?

Reply:
Ah, nevermind, it doesn't, I had just written the helper methods on `Queue` which took `&mut self` to the entire queue, when it only needed mutable access to the queue state. Moving the helper methods to the `QueueState` struct and adding an argument for the gl context fixed the borrowing issues without needing a mutex.

I pushed an update without the mutex.